src/platform/shell.go

```diff
@@ -785,60 +788,61 @@ func (env *Shell) Logs() string {
 
 func (env *Shell) TemplateCache() *TemplateCache {
 	defer env.Trace(time.Now())
-	if env.tmplCache != nil {
-		return env.tmplCache
-	}
-
-	tmplCache := &TemplateCache{
-		Root:         env.Root(),
-		Shell:        env.Shell(),
-		ShellVersion: env.CmdFlags.ShellVersion,
-		Code:         env.ErrorCode(),
-		WSL:          env.IsWsl(),
-		Segments:     make(map[string]interface{}),
-		PromptCount:  env.CmdFlags.PromptCount,
-	}
-	tmplCache.Env = make(map[string]string)
-	tmplCache.Var = make(map[string]interface{})
+	tmplCache := env.tmplCache
+	tmplCache.Lock()
+	defer tmplCache.Unlock()
+
+	if !tmplCache.initialized {
```

Comment:
can we invert this one to avoid having this whole block indented?

```go
if tmplCache.initialized {
    return tmplCache
}
// init here
```

Reply:
Yes, we can.